pkg/allocation/converters/converter_test.go

```diff
@@ -873,6 +875,7 @@ func TestConvertGSAToAllocationResponse(t *testing.T) {
 					Source:   "local",
 				},
 			},
+			grpcStatusCode: codes.OK,
```

Comment:
Again, this would never be `codes.OK` - it doesn't really make sense as an error return. What I would, though, is just omit this field and let it be 0 (which is incidentally the same as `codes.OK`, and then below add defaulting logic.